Projects/TMProject/SControl.cpp

```diff
@@ -1639,11 +1639,11 @@ int SCheckBox::OnMouseEvent(unsigned int dwFlags, unsigned int wParam, int nX, i
 
 	m_bOver = PointInRect(nX, nY, m_nPosX, m_nPosY, m_nWidth, m_nHeight);
 
-	if (dwFlags == 512)
+	if (dwFlags == 512 || dwFlags == 513)
 		return 0;
 
-	if (dwFlags != 513)
```

Comment:
Essa remoção não é necessária
512 refere-se a `WM_MOUSEMOVE`. Isto é, ignoramos movimentos de mouse.
513 refere-se a `WM_LBUTTONDOWN`, que é um clique do botão esquerdo.

Lendo o código, faz sentido que a gente ignore movimentos de mouse e cliques quando o `m_bOver != 1`. Sendo assim, tanto o código original, quanto o código descompilado quanto este fix parecem incorretos.

Minha sugestão seria:
1.  Fazer a mudança para ignorar cliques quando `m_bOver != 1`, bem parecido com o que você fez. Eu digo mudar para != 1 porque é o que normalmente o código faz.
2. Juntar o if do dwFlags != 513 com o if logo acima. sendo assim, sempre que for um clique, só aceitamos quando for no checkbox.

Sem o número dois, faça o seguinte teste. Insira o CheckBox numa janela qualquer e deixe aberto. Tente clicar em qualquer outro botão da janela, não vai funcionar o clique. Fechando a janela que possui o CheckBox, vai funcionar. O código final seria:
```c++
	if (dwFlags != 513 || m_bOver != 1)
		return SPanel::OnMouseEvent(dwFlags, wParam, nX, nY);
```

Reply:
Sobre o clique que é muito rápido, infelizmente a responsabilidade do tempo de clique é do controle em si. Como você pode ver no SButton, por exemplo, no OnMouseEvernt, ele mesmo tem a responsabilidade de checar o tempo para não clicar muito rapidamente. Você pode ver isto aqui:
```c++
			unsigned int dwServerTime = g_pTimerManager->GetServerTime();
			if (dwServerTime > m_dwOldTime + 500)
			{
```

O ideal seria fazer o mesmo no SCheckBox.